tests/cpp/common/test_json.cc

```diff
@@ -0,0 +1,360 @@
+    "debug_verbose": "0",
+    "disable_default_eval_metric": "0",
+    "dsplit": "auto",
+    "nthread": "0",
+    "seed": "0",
+    "seed_per_iteration": "0",
+    "test_flag": "",
+    "tree_method": "gpu_hist"
+  },
+  "configuration": {
+    "booster": "gbtree",
+    "n_gpus": "1",
+    "num_class": "0",
+    "num_feature": "10",
+    "objective": "reg:linear",
+    "predictor": "gpu_predictor",
+    "tree_method": "gpu_hist",
+    "updater": "grow_gpu_hist"
+  },
+  "objective": "reg:linear",
+  "booster": "gbtree",
+  "gbm": {
+    "GBTreeModelParam": {
+      "num_feature": "10",
+      "num_output_group": "1",
+      "num_roots": "1",
+      "size_leaf_vector": "0"
+    },
+    "trees": [{
+        "TreeParam": {
+          "num_feature": "10",
+          "num_roots": "1",
+          "size_leaf_vector": "0"
+        },
+        "num_nodes": "9",
+        "nodes": [
+          {
+            "depth": 0,
+            "gain": 31.8892,
+            "hess": 10,
+            "left": 1,
+            "missing": 1,
+            "nodeid": 0,
+            "right": 2,
+            "split_condition": 0.580717,
+            "split_index": 2
+          },
+          {
+            "depth": 1,
+            "gain": 1.5625,
+            "hess": 3,
+            "left": 5,
+            "missing": 5,
+            "nodeid": 2,
+            "right": 6,
+            "split_condition": 0.160345,
+            "split_index": 0
+          },
+          {
+            "depth": 2,
+            "gain": 0.25,
+            "hess": 2,
+            "left": 7,
+            "missing": 7,
+            "nodeid": 6,
+            "right": 8,
+            "split_condition": 0.62788,
+            "split_index": 0
+          },
+          {
+            "hess": 1,
+            "leaf": 0.375,
+            "nodeid": 8
+          },
+          {
+            "hess": 1,
+            "leaf": 0.075,
+            "nodeid": 7
+          },
+          {
+            "hess": 1,
+            "leaf": -0.075,
+            "nodeid": 5
+          },
+          {
+            "depth": 3,
+            "gain": 10.4866,
+            "hess": 7,
+            "left": 3,
+            "missing": 3,
+            "nodeid": 1,
+            "right": 4,
+            "split_condition": 0.238748,
+            "split_index": 1
+          },
+          {
+            "hess": 6,
+            "leaf": 1.54286,
+            "nodeid": 4
+          },
+          {
+            "hess": 1,
+            "leaf": 0.225,
+            "nodeid": 3
+          }
+        ],
+        "leaf_vector": []
+      }],
+    "tree_info": [0]
+  }
+}
+)json";
+  return model_json;
+}
+
+TEST(Json, TestParseObject) {
+  std::string str = R"obj({"TreeParam" : {"num_feature": "10"}})obj";
+  auto json = Json::Load(StringView{str.c_str(), str.size()});
+}
+
+TEST(Json, ParseNumber) {
+  std::string str = "31.8892";
+  auto json = Json::Load(StringView{str.c_str(), str.size()});
+  ASSERT_NEAR(get<JsonNumber>(json), 31.8892, kRtEps);
```

Comment:
I think we use float everywhere (not double) so I would suggest to add `f` suffix to the literal: 31.8892f
This is rather minor and nitpicking